arcgis-ios-sdk-samples/Edit data/Create and save KML file/CreateAndSaveKMLViewController.swift

```diff
@@ -0,0 +1,270 @@
+        alertController.addAction(cancelAction)
+        
+        alertController.popoverPresentationController?.barButtonItem = addButton
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    // Prompt color selection action sheet for polyline feature.
+    func addPolyline() {
+        let alertController = UIAlertController(title: "Select Color", message: "This color will be used for the polyline", preferredStyle: .actionSheet)
+        colors.forEach { (colorTitle, colorValue) in
+                let colorAction = UIAlertAction(title: colorTitle, style: .default) { (_) in
+                self.kmlStyle = self.makeKMLStyleWithLineStyle(color: colorValue)
+                self.startSketch(creationMode: .polyline)
+                }
+            alertController.addAction(colorAction)
+        }
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        
+        alertController.popoverPresentationController?.barButtonItem = addButton
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    // Prompt color selection action sheet for polygon feature.
+    func addPolygon() {
+        let alertController = UIAlertController(title: "Select Color", message: "This color will be used to fill the polygon", preferredStyle: .actionSheet)
+        colors.forEach { (colorTitle, colorValue) in
+                let colorAction = UIAlertAction(title: colorTitle, style: .default) { (_) in
+                self.kmlStyle = self.makeKMLStyleWithPolygonStyle(color: colorValue)
+                self.startSketch(creationMode: .polygon)
+                }
+            alertController.addAction(colorAction)
+        }
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        
+        alertController.popoverPresentationController?.barButtonItem = addButton
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    // Make KML with a point style.
+    func makeKMLStyleWithPointStyle(iconURL: URL) -> AGSKMLStyle {
+        let icon = AGSKMLIcon(url: iconURL)
+        let iconStyle = AGSKMLIconStyle(icon: icon, scale: 1.0)
+        kmlStyle.iconStyle = iconStyle
+        return kmlStyle
+    }
+    
+    // Make KML with a line style.
+    func makeKMLStyleWithLineStyle(color: UIColor) -> AGSKMLStyle {
+        kmlStyle.lineStyle = AGSKMLLineStyle(color: color, width: 2.0)
+        return kmlStyle
+    }
+    
+    // Make KML with a polygon style.
+    func makeKMLStyleWithPolygonStyle(color: UIColor) -> AGSKMLStyle {
+        kmlStyle.polygonStyle = AGSKMLPolygonStyle(color: color)
+        kmlStyle.polygonStyle?.isFilled = true
+        kmlStyle.polygonStyle?.isOutlined = false
+        return kmlStyle
+    }
```

Comment:
These methods are named `make` and return a style, but they don't actually make anything. Please modify them to actually create the KML style.

Reply:
> Please modify them to actually create the KML style.

Are you suggesting I adjust the function names or to create a new KML style within each function?

Reply:
I suggest creating the KML style in each method.